src/Components/InputModel.re

```diff
@@ -1,5 +1,33 @@
 open Oni_Core;
 
+let wordSeparators = " ./\\()\"'-:,.;<>~!@#$%^&*|+=[]{}`~?";
+
+let separatorOnIndex = (text, index) => {
+  String.contains(wordSeparators, String.get(text, index));
+}
+
+let findNextWordBoundary = (text, cursorPosition) => {
+  let finalIndex = String.length(text);
+  let index = ref(min(cursorPosition + 1, finalIndex));
+
+  while (index^ < finalIndex && !separatorOnIndex(text, index^)) {
+    index:= index^ + 1;
+  };
+
+  index^;
+};
+
+let findPrevWordBoundary = (text, cursorPosition) => {
+  let finalIndex = 0;
+  let index = ref(max(cursorPosition - 1, finalIndex));
+
+  while (index^ > finalIndex && !separatorOnIndex(text, index^ - 1)) {
+    index:= index^ - 1;
+  };
+
+  index^;
+};
```

Comment:
It would probably be better to use `Str.search_forward` and `Str.search_backward` for this. Or if that doesn't fit or perform well enough, generalize and extract this into `Utility.StringEx`.

Reply:
So this is really big question. I'd better leave it for now here because it's not handling Unicode right no. Current implementations will work with Unicode easily.
This something I really what to discuss and work on it after this PR.

In short all this thing should tolerate Unicode since filenames, search queries, etc. might be in non english (Ukrainian, Chinese, Japanise,  German..)

I'd better Create an Issue about Updating this to handle Unicode and own it.

Reply:
That would be a huge help @fanantoxa ! We're missing proper Unicode support (and some other stuff, like font fallbacks) in those places. 

I believe the [Oniguruma](https://github.com/kkos/oniguruma) regex engine we use can support Unicode (and we have an OCaml/Reason binding for it here: https://github.com/onivim/reason-oniguruma) - so could be an option to think about later too.